bundles/org.openhab.binding.emotiva/src/main/java/org/openhab/binding/emotiva/internal/EmotivaProcessorHandler.java

```diff
@@ -0,0 +1,734 @@
+        EnumMap<EmotivaControlCommands, String> bands = new EnumMap<>(
+                Map.of(band_am, band_am.getLabel(), band_fm, band_fm.getLabel()));
+        modes = new EnumMap<>(EmotivaSubscriptionTags.class);
+        commandMaps.put(NAME_SOURCES_MAP, sources);
+        commandMaps.put(tuner_channel.getName(), channels);
+        commandMaps.put(tuner_band.getName(), bands);
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("initialize: {}", getThing().getUID());
+
+        scheduler.execute(this::connect);
+    }
+
+    private synchronized void connect() {
+        final EmotivaConfiguration localConfig = config;
+        try {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "@text/message.processor.connecting");
+
+            final EmotivaUdpReceivingService notifyListener = new EmotivaUdpReceivingService(localConfig.notifyPort,
+                    localConfig, scheduler);
+            this.notifyListener = notifyListener;
+            notifyListener.connect(this::handleStatusUpdate, true);
+
+            final EmotivaUdpSendingService sendConnector = new EmotivaUdpSendingService(localConfig, scheduler);
+            sendingService = sendConnector;
+            sendConnector.connect(this::handleStatusUpdate, true);
+
+            for (int attempt = 1; attempt <= CONNECTION_RETRIES && !udpSenderActive; attempt++) {
+                try {
+                    logger.debug("Attempt {}", attempt);
+                    sendConnector.sendSubscription(generalSubscription, config);
+                    sendConnector.sendSubscription(nonGeneralSubscriptions, config);
+                    // TODO Request bar_update for volume, will give user specified max volume
+
+                } catch (IOException e) {
+                    // network or socket failure, also wait 2 sec and try again
+                }
+
+                // answer expected within 50-600ms on a regular network; wait up to 2sec just to make sure
+                for (int delay = 0; delay < 10 && !udpSenderActive; delay++) {
+                    Thread.sleep(200); // wait 10 x 200ms = 2sec
+                }
+            }
+
+            if (udpSenderActive) {
+                updateStatus(ThingStatus.ONLINE);
+
+                final EmotivaUdpReceivingService menuListenerConnector = new EmotivaUdpReceivingService(
+                        localConfig.menuNotifyPort, localConfig, scheduler);
+                this.menuNotifyListener = menuListenerConnector;
+                menuListenerConnector.connect(this::handleStatusUpdate, true);
+
+                startPollingKeepAlive();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "@text/message.processor.connection.failed");
+                disconnect();
+                scheduleConnectRetry(retryConnectInMinutes);
+            }
```

Comment:
Not sure, the connection method is synchronized and try to make sure the connection is either working or we need to schedule a retry. Suggestions are welcome.

Reply:
Regarding the ThingStatus, it is a common practice to:
- In `initialize()`
    - Set `updateStatus(ThingStatus.UNKNOWN);` at the start
    - Check configuration and set `ThingStatus` with details if it is not valid
    - End with a scheduled task (e.g. `updateTask()`) that updates the thing state and channels
- In `updateTask()`
    - Check if connection is authenticated and continue or get authenticated and set `ThingStatus`
    - Get data to update channels, set `ThingStatus` on error.

Not covering all cases and when applied to this binding it might not suit 100%.

Reply:
Updated initialize() with status setting and checking of port and ipaddress to the device.